src/vaults/BalanceSheet.sol

```diff
@@ -39,8 +39,13 @@ contract BalanceSheet is Auth, Recoverable, IBalanceSheet, IBalanceSheetGatewayH
     ISharePriceProvider public sharePriceProvider;
 
     mapping(PoolId => mapping(ShareClassId => mapping(address => bool))) public manager;
-    mapping(PoolId poolId => mapping(ShareClassId scId => int128 amount)) public queuedShares;
-    mapping(PoolId poolId => mapping(ShareClassId scId => mapping(AssetId assetId => int128))) public queuedAssets;
+
+    mapping(PoolId poolId => mapping(ShareClassId scId => uint128 amount)) public queuedShareIssuances;
+    mapping(PoolId poolId => mapping(ShareClassId scId => uint128 amount)) public queuedShareRevocations;
```

Comment:
We should put these into 1 struct like
```solidity
struct QueuedShares {
    uint128 issuances;
    uint128 revocations;
}
```

Since they are read and written at the same time so this removes 1 SLOAD + SSTORE.

Same for `queuedAssetDeposits/queuedAssetWithdraws `.